### PR TITLE
Export & deploy updates (part II)

### PR DESCRIPTION
**Pull request by janekl**
# What does this PR do ?

Several updates & bugfixes for `nemo.export` module.
* Fixing `UnboundLocalError` when building tokenizer in `nemo/export/trt_llm/nemo_ckpt_loader/nemo_file.py`
* Making vLLM deployment consistent with TRT-LLM as for `output_generation_logits` (disabled in case of vLLM for now)
* Using `BlockManagerV2` by default for vLLM deployment (as `V1` is getting deprecated)
* Listing all (minimal) requirements to run vLLM in `/opt/venv` in the dedicated file `requirements_vllm.txt.`
* Other minor changes

**Collection**: NLP / LLM

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: requirements/requirements_multimodal.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ diffusers>=0.19.3
 einops_exts
 imageio
 kornia
-megatron-energon
+megatron-energon<3.0.0
```
Comment (janekl) [inline requirements/requirements_multimodal.txt:8]: FYI @ko3n1g

**Review by oyilmaz-nvidia — APPROVED** [review]
LGTM.
